source/common/common/logger.cc

```diff
@@ -254,6 +254,21 @@ void Registry::setLogFormat(const std::string& log_format) {
   }
 }
 
+ProtobufUtil::Status Registry::setJsonLogFormat(const Protobuf::Message& log_format_struct) {
+  Protobuf::util::JsonPrintOptions json_options;
+  json_options.preserve_proto_field_names = true;
+  json_options.always_print_primitive_fields = true;
+
+  std::string format_as_json;
+  auto status =
+      Protobuf::util::MessageToJsonString(log_format_struct, &format_as_json, json_options);
+  if (status.ok()) {
+    setLogFormat(format_as_json);
```

Comment:
I think the logged message may not be a valid JSON escaped string. That is, the pattern will be, but the logged message may not.
Will it make sense to replace/reject a config with `%_`?

Reply:
Sorry but I'm not sure if I understand what you mean. "%_" stands for "The actual message to log, but with escaped newlines". If the schema log schema is:

```
application_log_format:
  json_format:
    Message: "%_"
```

Wouldn't the outcome be `{"Message":"message that was escaped\nhere"}`?

Did you (maybe) mean that we should enforce that "%v" is not used? Because %v stands for "The actual message to log", so for valid JSON format the message couldn't have newlines in it?